### PR TITLE
Ensure EventListenerGuard in every partition

### LINKED ISSUE #10680: Test partitioning can eliminate EventListener race workaround
http://dotnet-ci.cloudapp.net/job/roslyn_master_win_rel_unit64/1112/artifact/Binaries/Release/xUnitResults/Roslyn.VisualStudio.Services.UnitTests.dll.0.out.log

System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at System.Diagnostics.Tracing.EventListener.DisposeOnShutdown(Object sender, EventArgs e)

http://dotnet-ci.cloudapp.net/job/roslyn_master_win_dbg_unit64/1111/artifact/Binaries/Debug/xUnitResults/Roslyn.Services.Editor.UnitTests2.dll.0.out.log

System.NullReferenceException: Object reference not set to an instance of an object.
   at Xunit.Sdk.TestRunner`1.<RunAsync>d__43.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestCaseRunner`1.<RunAsync>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestMethodRunner`1.<RunTestCasesAsync>d__32.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestMethodRunner`1.<RunAsync>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestClassRunner`1.<RunTestMethodsAsync>d__38.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestClassRunner`1.<RunAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestCollectionRunner`1.<RunTestClassesAsync>d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestCollectionRunner`1.<RunAsync>d__27.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestAssemblyRunner`1.<RunTestCollectionsAsync>d__42.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.XunitTestAssemblyRunner.<RunTestCollectionsAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestAssemblyRunner`1.<RunAsync>d__41.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.XunitTestFrameworkExecutor.<RunTestCases>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_1(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()

### PR DESCRIPTION
**Pull request by jaredpar**
EventListenerGuard is a necessary protection from [CoreFX bug 3793](dotnet/corefx#3793).  When partitioning up assemblies we need to ensure that this type is present in every test partition.

closes #10680

### CODE REVIEW

**Comment by jaredpar** [comment]
The actual bug fix is in this commit:

> https://github.com/dotnet/roslyn/pull/10682/commits/f344308881a6d15ca125615db337563b5a45cc0a

The other commit is doing a rename chunk -> partition.  This is the verbiage we use to describe this now.  May as well fix it while I'm in there.  

CC @dotnet/roslyn-ide

File: src/Tools/Source/RunTests/AssemblyScheduler.cs [hunk 1]
```diff
@@ -44,6 +44,16 @@ internal AssemblyInfo(string assemblyPath, bool useHmtl)
...
     {
+        /// <summary>
+        /// This is a test class inserted into assemblies to guard against a .NET desktop bug.  The tests
+        /// inside of it counteract the underlying issue.  If this test is included in any assembly it 
+        /// must be added to every partition to ensure the work around is present
+        /// 
+        /// https://github.com/dotnet/corefx/issues/3793
+        /// https://github.com/dotnet/roslyn/issues/8936
+        /// </summary>
+        private const string EventListenerGuardFullName = "Microsoft.CodeAnalysis.UnitTests.EventListenerGuard";
```
Comment (jasonmalinowski) [inline src/Tools/Source/RunTests/AssemblyScheduler.cs]: Do we have any protection if this type were to get renamed?

File: src/Tools/Source/RunTests/Program.cs [hunk 2]
```diff
@@ -220,7 +220,8 @@ private static IOrderedEnumerable<string> OrderAssemblyList(IEnumerable<string>
             obj["Is32Bit"] = !options.Test64;
             obj["AssemblyCount"] = options.Assemblies.Count;
             obj["CacheCount"] = result.CacheCount;
-            obj["ChunkCount"] = chunkCount;
+            obj["ChunkCount"] = partitionCount;
```
Comment (jasonmalinowski) [inline src/Tools/Source/RunTests/Program.cs]: Comment why the duplication?

**Comment by jasonmalinowski** [comment]
:+1:

File: src/Tools/Source/RunTests/Program.cs (hunk 2 above)
Comment (jaredpar) [inline src/Tools/Source/RunTests/Program.cs]: Arg.  Meant to change the name on that use.  Will update.

File: src/Tools/Source/RunTests/AssemblyScheduler.cs (hunk 1 above)
Comment (jaredpar) [inline src/Tools/Source/RunTests/AssemblyScheduler.cs]: No.  Open to ideas on protecting it.  Can't really see a way of doing it though.

File: src/Tools/Source/RunTests/Program.cs (hunk 2 above)
Comment (jaredpar) [inline src/Tools/Source/RunTests/Program.cs]: NM on the duplication.  Thie code is correct.  I want to transition the name and need old + new until the code makes its way through the branches and webiset.

**Pull request merged**